src/TesApi.Web/AzureProxy.cs

```diff
@@ -290,9 +289,94 @@ public async Task DeleteBatchJobAsync(PoolInformation poolInformation, Cancellat
             await batchClient.JobOperations.DeleteJobAsync(poolInformation.PoolId, cancellationToken: cancellationToken);
         }
 
+        /// <inheritdoc/>
+        public async Task<(IReadOnlyDictionary<CloudPool, IReadOnlyList<ComputeNode>> PoolsAndNodes, IReadOnlyDictionary<CloudJob, IReadOnlyList<CloudTask>> JobsAndTasks)> GetBatchAccountStateAsync(bool usingAutoPools, IEnumerable<string> ids, CancellationToken cancellationToken)
+        {
+            List<CloudJob> jobs = default;
+            IAsyncEnumerable<CloudPool> pools = default;
+
+            var idList = ids.ToList();
+
+            var jobFilter = new ODATADetailLevel
+            {
+                SelectClause = "id,state,executionInfo,creationTime"
+            };
+
+            var poolFilter = new ODATADetailLevel
+            {
+                SelectClause = "id,resizeErrors"
+            };
+
+            var nodeFilter = new ODATADetailLevel
+            {
+                SelectClause = "id,state,startTaskInfo,errors"
+            };
+
+            var taskFilter = new ODATADetailLevel
+            {
+                SelectClause = "id,state,executionInfo"
+            };
+
+            var idStartsWith = new Func<string, string, bool>((id, format) => ids.Select(i => string.Format(format, i)).Any(s => id.StartsWith(s)));
+
+            if (usingAutoPools)
+            {
+                jobs = await CachingWithRetriesAzureProxy.asyncRetryPolicy.ExecuteAsync(() =>
+                        batchClient.JobOperations.ListJobs(jobFilter).ToAsyncEnumerable(),
+                        CachingWithRetriesAzureProxy.retryPolicy)
+                    .Where(j => idStartsWith(j.Id, $"{{0}}{BatchJobAttemptSeparator}"))
+                    .ToListAsync(cancellationToken);
+
+                var poolIds = jobs.Select(j => j.ExecutionInformation?.PoolId).Where(s => !string.IsNullOrWhiteSpace(s)).ToList();
+
+                pools = CachingWithRetriesAzureProxy.asyncRetryPolicy.ExecuteAsync(() =>
+                        batchClient.PoolOperations.ListPools(poolFilter).ToAsyncEnumerable(),
+                        CachingWithRetriesAzureProxy.retryPolicy)
+                    .Where(p => poolIds.Contains(p.Id));
+            }
+            else
+            {
+                jobs = await CachingWithRetriesAzureProxy.asyncRetryPolicy.ExecuteAsync(() =>
+                        batchClient.JobOperations.ListJobs(jobFilter).ToAsyncEnumerable(),
+                        CachingWithRetriesAzureProxy.retryPolicy)
+                    .Where(j => idList.Contains(j.Id))
+                    .ToListAsync(cancellationToken);
+
+                pools = CachingWithRetriesAzureProxy.asyncRetryPolicy.ExecuteAsync(() =>
+                        batchClient.PoolOperations.ListPools(poolFilter).ToAsyncEnumerable(),
+                        CachingWithRetriesAzureProxy.retryPolicy)
+                    .Where(p => idList.Contains(p.Id));
```

Comment:
Any potential scale concerns here?

Reply:
Fewer calls will be faster than smaller calls, if all the information is eventually needed.